src/runtime/sgx/untrusted/sgx_module.cc

```diff
@@ -0,0 +1,143 @@
+/*!
+ *  Copyright (c) 2018 by Contributors
+ * \file cur_module.cc
+ * \brief SGX enclave module.
+ */
+#include <dmlc/logging.h>
+#include <tvm/runtime/c_runtime_api.h>
+#include <tvm/runtime/registry.h>
+#include <tvm/runtime/threading_backend.h>
+#include <sgx_urts.h>
+#include <fstream>
+#include "../common.h"
+#include "../../file_util.h"
+
+namespace tvm {
+namespace runtime {
+
+class SGXModuleNode;
+
+namespace sgx {
+  thread_local SGXModuleNode* cur_mod;
```

Comment:
Better not to introduce thread local global variable. Always put static before thread_local.

Introduce SGXModuleContext(SGXModuleNode*), and use RAII to set and remove current context. Have a static function to query CurrentContext and CHECK that the current module cannot be nullptr

Reply:
So you're proposing to have an explicit global variable instead? There's going to have to be a global either way. This was just the solution recommended on the [SGX SDK issue tracker](https://github.com/intel/linux-sgx/issues/226).